js/ui/appDisplay.js

```diff
@@ -772,7 +803,14 @@ const AllView = new Lang.Class({
 
     _redisplay: function() {
         if (this.getAllItems().length == 0) {
-            this.addIcons();
+            if (Main.layoutManager.startingUp) {
+                this.addIcons(true);
+
+                Main.layoutManager.connect('startup-complete',
+                                           Lang.bind(this, this._animateIconsIn));
```

Comment:
As mentioned in https://phabricator.endlessm.com/T18012#513379, this is causing problems on startup because startup-complete is too late to do the animation, since Overview's `showApps()` method is already called on startup-prepare.

Maybe call `_animateIconsIn()` rightaway from here?